reactor-netty-core/src/main/java/reactor/netty/transport/ProxyProvider.java

```diff
@@ -173,6 +176,17 @@ public void addProxyHandler(Channel channel) {
 		Objects.requireNonNull(channel, "channel");
 		ChannelPipeline pipeline = channel.pipeline();
 		pipeline.addFirst(NettyPipeline.ProxyHandler, newProxyHandler());
+		// For SOCKS proxy, the netty SOCKS handlers may register listeners in channel promises, so we need to register a
+		// special handler which ensures that any VoidPromise will be converted to "unvoided" promises (for support of listeners).
+		// Note: an example of a VoidPromise which does not support listeners is the MonoSendMany.SendManyInner promise.
+		if (this.type == Proxy.SOCKS4 || type == Proxy.SOCKS5) {
+			pipeline.addAfter(NettyPipeline.ProxyHandler, NettyPipeline.UnvoidHandler, new ChannelOutboundHandlerAdapter() {
+				@Override
+				public void write(ChannelHandlerContext ctx, Object msg, ChannelPromise promise) throws Exception {
+					ctx.write(msg, promise.unvoid());
+				}
```

Comment:
nit: Let's fix the warnings


```suggestion
				@Override
				@SuppressWarnings("FutureReturnValueIgnored")
				public void write(ChannelHandlerContext ctx, Object msg, ChannelPromise promise) {
					//"FutureReturnValueIgnored" this is deliberate
					ctx.write(msg, promise.unvoid());
				}
```